src/effects/effects_base/internal/effectexecutionscenario.cpp

```diff
@@ -243,9 +244,14 @@ muse::Ret EffectExecutionScenario::doPerformEffect(au3::Au3Project& project, con
     //! NOTE Step 6 - perform effect
     //! ============================================================================
     // common things used below
-    const Ret success = numSelectedClips > 1
-                        ? performEffectOnEachSelectedClip(project, *effect, pInstanceEx, *settings)
-                        : effectsProvider()->performEffect(project, effect, pInstanceEx, *settings);
+    Ret success;
+    if (numSelectedClips == 0) {
+        success = effectsProvider()->performEffect(project, effect, pInstanceEx, *settings);
```

Comment:
I think it would be better to add a function with an appropriate name, or a comment that we are trying to apply the effect to the selected range. 
(otherwise, if not in context, it is difficult to understand what is happening here)